### ISSUE TITLE
Automap lines may flicker

### ISSUE THREAD

**Issue by MrAlaux** [fabiangreffrath/woof#1580]
_Tested in the [latest master artifact (x64)](https://github.com/fabiangreffrath/woof/actions/runs/8222984635)._

Steps to reproduce, with a fresh config file:

- Disable fullscreen and _Dynamic Resolution_, set _Resolution Scale_ to _Native_, and disable _Smooth Pixel Scaling_
- Shrink the window as much as possible
- Enter the Automap
- Move around

You should see some of the Automap lines flickering in and out. A little preview:

<details>
  <summary>Image</summary>
  
![woof0001](https://github.com/fabiangreffrath/woof/assets/73968015/c4057172-3c48-4645-93fc-0d7741109617)
</details>

I believe that this occurs because said lines are 1-game-pixel wide, and when the window is downscaled, some pixels are lost. At least with my 1366x768 display, Smooth Pixel Scaling is good enough to prevent the lines from fully disappearing, although they can still become a bit hard to see.

I had already faced this problem in Nugget, and I put a dirty hack in place to mitigate it: thickening the lines if the screen were downscaled and Smooth Pixel Scaling were disabled. Perhaps such a thing can be applied here, hopefully with a better execution.

**Comment by rfomin** [comment]
> - Disable fullscreen and Dynamic Resolution, set Resolution Scale to Native, and disable Smooth Pixel Scaling
> - Shrink the window as much as possible

We can solve this particular case by changing the resolution when resizing the window, but I don't like it. Why set a high resolution and make a tiny window? I don't think we should worry about such cases.

**Comment by MrAlaux** [comment]
> Why set a high resolution and make a tiny window? I don't think we should worry about such cases.

Personally, sometimes I disable fullscreen and play windowed for a bit. Fair point, though.

**Comment by ceski-1** [comment]
It seems like native uses the monitor resolution instead of the window size, so it ends up downsampling (or "supersampling") and causing these artifacts. If native used the window size instead, it would probably fix the issue.

**Comment by rfomin** [comment]
> It seems like native uses the monitor resolution instead of the window size, so it ends up downsampling (or "supersampling") and causing these artifacts. If native used the window size instead, it would probably fix the issue.

Right, I've tried the window size for native resolution, it doesn't look good in practice. "Supersampling" may be a legit option, but I don't think our rendering is fast enough for that.

**Comment by fabiangreffrath** [comment]
How about this?

```patch
--- a/src/i_video.c
+++ b/src/i_video.c
@@ -1334,7 +1334,10 @@ static void ResetLogicalSize(void)
     else
     {
         DestroyUpscaledTexture();
-        SDL_SetTextureScaleMode(texture, SDL_ScaleModeNearest);
+        if (window_width < video.width || window_height < video.height)
+            SDL_SetTextureScaleMode(texture, SDL_ScaleModeLinear);
+        else
+            SDL_SetTextureScaleMode(texture, SDL_ScaleModeNearest);
     }
 }
 
```

**Comment by MrAlaux** [comment]
We now have the automap line thickness setting, which helps mitigate this.

**Comment by elf-alchemist** [comment]
@MrAlaux did you try the patch above? I doubt it is 100% the same with SDL3, now, but maybe something like it still works?

**Comment by MrAlaux** [comment]
> [@MrAlaux](https://github.com/MrAlaux) did you try the patch above? I doubt it is 100% the same with SDL3, now, but maybe something like it still works?

I just hacked something together to try it, and it does help a bit, but it's not a definitive solution.

Since we have adjustable line thickness now, I suppose a better solution would be to adjust said thickness if the window size is lesser than the "render" size.

**Comment by fabiangreffrath** [comment]
Right, I think we didn't have Automap line thickness back when I proposed the patch above. There should be some trigger to temporarily increase the config-given line thickness if windows size is decreased below resolution.
